march_hardware_builder/test/joint_builder_test.cpp

```diff
@@ -40,20 +40,18 @@ TEST_F(JointTest, ValidJointHip)
   this->joint->safety->soft_lower_limit = 0.1;
   this->joint->safety->soft_upper_limit = 1.9;
 
-  march::Joint created = HardwareBuilder::createJoint(config, "test_joint_hip", this->joint);
+  const std::string name = "test_joint_hip";
+  march::Joint created = HardwareBuilder::createJoint(config, name, this->joint);
+
+  auto absolute_encoder = std::make_unique<march::AbsoluteEncoder>(
+      16, 22134, 43436, this->joint->limits->lower, this->joint->limits->upper, this->joint->safety->soft_lower_limit,
```

Comment:
Store these values in a parameter since you use them often?

Reply:
These values are from the test yaml and are only valid here, so I don't think it's necessary to turn them into parameters

Reply:
Why not? That way, you will only have to change the value once when it changes in the yaml, and it is also clearer what these numbers stand for, since you would have to go look for the constructor to find out the first is bits, the second and third limits in IU.

Reply:
Okay, so something like this?
```cpp
TEST_F(JointTest, ValidNotActuated)
{
  YAML::Node config = this->loadTestYaml("/joint_correct_not_actuated.yaml");
  this->joint->limits->lower = 0.0;
  this->joint->limits->upper = 2.0;
  this->joint->safety->soft_lower_limit = 0.1;
  this->joint->safety->soft_upper_limit = 1.9;
  const std::string name = "test_joint_hip";
  const size_t abs_encoder_resolution = 16;
  const int32_t abs_encoder_lower_limit = 22134;
  const int32_t abs_encoder_upper_limit = 43436;
  const size_t inc_encoder_resolution = 12;
  const double inc_encoder_transmission = 50.0;
  const int slave_index = 1;
  const int byte_offset = 2;
  const int net_number = -1;
  const bool allow_actuation = false;

  march::Joint created = HardwareBuilder::createJoint(config, name, this->joint);

  auto absolute_encoder = std::make_unique<march::AbsoluteEncoder>(
      abs_encoder_resolution, abs_encoder_lower_limit, abs_encoder_upper_limit, this->joint->limits->lower,
      this->joint->limits->upper, this->joint->safety->soft_lower_limit, this->joint->safety->soft_upper_limit);
  auto incremental_encoder =
      std::make_unique<march::IncrementalEncoder>(inc_encoder_resolution, inc_encoder_transmission);
  auto imc = std::make_unique<march::IMotionCube>(slave_index, std::move(absolute_encoder),
                                                  std::move(incremental_encoder), march::ActuationMode::unknown);
  auto ges = std::make_unique<march::TemperatureGES>(slave_index, byte_offset);
  march::Joint expected(name, net_number, allow_actuation, std::move(imc), std::move(ges));

  ASSERT_EQ(expected, created);
}
```

Reply:
No, I meant put them in the setup since you use these values in most of the tests. The yaml is also loaded there so it makes sense to keep these together.